### PR TITLE
Update target versions

### PR DESCRIPTION
**Pull request by astj**
Current target (Chef 13, 14, Ruby 2.3, 2.4) all reached EOL.

### CODE REVIEW

File: docker/amazonlinux-2/Dockerfile [hunk 1]
```diff
@@ -17,4 +17,4 @@ RUN echo '{"run_list":["recipe[mackerel-agent]"], "mackerel-agent": {"start_on_s
 COPY ./cookbooks /cookbooks
 
 # Hack for Chef 14. yum_package fails with EOFError without redirecting stdio...
-RUN chef-client -j /var/chef/node/localhost.json | cat
+RUN chef-client --chef-license accept-silent -j /var/chef/node/localhost.json | cat
```
Comment (astj) [inline docker/amazonlinux-2/Dockerfile:20]: On Chef >= 15, We need to accept EULA https://docs.chef.io/chef_license_accept/

File: docker/debian-9/Dockerfile [hunk 2]
```diff
@@ -2,8 +2,8 @@ FROM debian:9
 
 ARG chefver
 RUN apt-get update && apt-get install -y curl gnupg && \
-    curl -sL -o /tmp/chef.deb \
-        https://packages.chef.io/files/stable/chef/${chefver}/debian/8/chef_${chefver}-1_amd64.deb && \
+    curl -fsL -o /tmp/chef.deb \
```
Comment (astj) [inline docker/debian-9/Dockerfile:5]: add -f to fail immediately on failure

File: docker/debian-9/Dockerfile [hunk 3]
```diff
@@ -2,8 +2,8 @@ FROM debian:9
...
+        https://packages.chef.io/files/stable/chef/${chefver}/debian/9/chef_${chefver}-1_amd64.deb && \
```
Comment (astj) [inline docker/debian-9/Dockerfile:6]: somehow it was 8… (maybe, older Chef did not provide deb for Debian 9 and we used deb for Debian 8 instead)

**Review by yseto — APPROVED** [review]
seems good.

**Pull request merged**